crates/bevy_animation/src/lib.rs

```diff
@@ -942,26 +985,35 @@ fn apply_keyframe(
     }
 }
 
-fn update_transitions(player: &mut AnimationPlayer, time: &Time) {
-    player.transitions.retain_mut(|animation| {
-        animation.current_weight -= animation.weight_decline_per_sec * time.delta_seconds();
-        animation.current_weight > 0.0
-    });
+impl AnimationTargetId {
+    /// Creates a new [`AnimationTargetId`] by hashing a list of names.
+    ///
+    /// Typically, this will be the path from the animation root to the
+    /// animation target (bone) that is to be animated.
+    pub fn from_names<'a>(names: impl Iterator<Item = &'a Name>) -> Self {
+        let mut sha1 = Sha1::new();
+        sha1.update(ANIMATION_TARGET_NAMESPACE.as_bytes());
+        names.for_each(|name| sha1.update(name.as_bytes()));
+        let hash = sha1.digest().bytes()[0..16].try_into().unwrap();
+        Self(*uuid::Builder::from_sha1_bytes(hash).as_uuid())
+    }
+
+    /// Creates a new [`AnimationTargetId`] by hashing a single name.
+    pub fn from_name(name: &Name) -> Self {
```

Comment:
`impl From<&Name> for AnimationTargetId` rather?

Reply:
I don't mind the method, it just feels more idiomatic to use `From` in that case. It's a minor thing.

Reply:
I added the `From` impl.